custom/icds/translations/integrations/transifex.py

```diff
@@ -39,22 +39,20 @@ def __init__(self, domain, app_id, source_lang, project_slug, version=None, lang
         self.source_lang = source_lang
         self.lock_translations = lock_translations
         self.use_version_postfix = use_version_postfix
-        self.po_file_generator = POFileGenerator(domain, app_id, version, self.key_lang, source_lang, lang_prefix,
-                                                 exclude_if_default, use_version_postfix)
+        self.transifex_po_file_generator = TransifexPOFileGenerator(domain, app_id, version, self.key_lang,
+                                                                    source_lang, lang_prefix, exclude_if_default,
+                                                                    use_version_postfix)
 
     def send_translation_files(self):
         """
         submit files to transifex for performing translations
         """
         try:
-            self.po_file_generator.generate_translation_files()
+            self.transifex_po_file_generator.generate_translation_files()
             file_uploads = self._send_files_to_transifex()
+        finally:
             self._cleanup()
-            return file_uploads
-        except:
-            t, v, tb = sys.exc_info()
-            self._cleanup()
-            six.reraise(t, v, tb)
+        return file_uploads
```

Comment:
**[non-blocking]**
I think you commented on this somewhere, but this is another example of what @millerdev was highlighting.  `TransifexPOFileGenerator` has 4 interactions with this client code - initialization, generation, file access, and cleanup.  What makes this confusing is that they are all order-dependent.  You'd get an error if you attempted to do any of that out of order, and it wouldn't make sense to do any less than all 4 operations.  For that reason, my preference would be to write the interface accordingly.

This means one step which accepts all necessary parameters, generates the files, and returns the result.  That could be done with a simple function, but it still leaves cleanup.  Cleanup is something that needs to always happen, and you can trust the calling code to handle it in a `finally` block like you've done here, but python's context managers are meant specifically for this pattern.  The way I think of it, is that you're allocating resources, which should be signified with a `with` statement that'll handle allocation and deallocation cleanly and automatically.

```python
with TransifexPoFileGenerator(...) as po_files:
    # do something with po_files
```

If implementing that is more trouble than it's worth, you could also opt to constrain `transifex_po_file_generator` to just one method so you can see it's whole lifecycle:

```python
def send_translation_files(self):
    try:
        transifex_po_file_generator = TransifexPoFileGenerator(...)
        transifex_po_file_generator.generate_translation_files()
        file_uploads = self._send_files_to_transifex(
            transifex_po_file_generator.generated_files)
    finally:
        transifex_po_file_generator._cleanup()
    return file_uploads
```
This way nothing outside of that method needs to know about `transifex_po_file_generator`, and you don't have to worry about multiple methods accessing a mutating shared state.

That's all the pontificating I'm gonna do on the topic, thank you for your patience.  I'm not expecting you to act on these suggestions, I'm more thinking of this as food for thought for next time, and options if you end up deciding to refactor some of this anyways.  Hope it's helpful.